### PR TITLE
refactor: replace p-retry with custom implementation based on retry

### PR DESCRIPTION
**Pull request by idoshamun**
Since we're using an old p-retry version and it's not possible to upgrade at this point we decided to roll our own based on the retry package (same as p-retry)

### CODE REVIEW

File: src/integrations/retry.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import retry, { OperationOptions } from 'retry';
```
Comment (rebelchris) [inline src/integrations/retry.ts:1]: I see you based it on a earlier version, did we not want to stick to the latest one?

File: src/integrations/retry.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+          if (!operation.retry(err)) {
+            throw operation.mainError();
+          }
+        } catch (finalError) {
+          reject(finalError);
+        }
+      }
+    });
+  });
```
Comment (capJavert) [inline src/integrations/retry.ts:66]: Should we do some tests for this?

File: src/integrations/retry.ts (hunk 1 above)
Comment (idoshamun) [reply src/integrations/retry.ts:1]: This is the latest one actually: https://github.com/tim-kos/node-retry/blob/master/package.json#L7

Comment (rebelchris) [reply src/integrations/retry.ts:1]: Ah I compared to this? https://github.com/sindresorhus/p-retry/blob/main/index.js

File: src/integrations/retry.ts (hunk 2 above)
Comment (idoshamun) [reply src/integrations/retry.ts:66]: Good point! Added

File: src/integrations/retry.ts (hunk 1 above)
Comment (idoshamun) [reply src/integrations/retry.ts:1]: Yes that's what I thought :)
So we're all clear I guess

**Review by rebelchris — APPROVED** [review]

**Pull request merged**